### PR TITLE
Checkout process

### PR DESCRIPTION
**Pull request by anandsinghparihar**
- [x] Refactor bid process after bid any item change tab cart to Buy

Creates separate component for
- [x] CART: cartsComponents for manage cart process.
- [ ] Shipping Address functionality. 
- [ ] ORDERS: ordersComponent for display all purchases item or sale item.

### CODE REVIEW

**Comment by HenkSwardt** [comment]
https://github.com/particl/particl-desktop/issues/932

**Comment by HenkSwardt** [comment]
https://github.com/particl/particl-desktop/issues/931

**Comment by vikas-cis** [comment]
#935

**Comment by vikas-cis** [comment]
@anandsinghparihar can you also check if order is placed then cart needs to be empty ?

THanks

File: src/app/core/market/api/bid/bid.service.ts [hunk 1]
```diff
@@ -33,12 +29,12 @@ export class BidService {
               (res) => {
                 this.log.d(`Bid placed for hash=${listing.hash} shipping to addressId=${addressIdOfProfile}`);
                 if (++nBidsPlaced === cart.listings.length) {
-                  this.snackbar.open('Order has been successfully placed');
-                  resolve(true);
+                  observer.next(true);
+                  observer.complete();
                 }
               },
               (failed) => {
```
Comment (rynomster) [inline src/app/core/market/api/bid/bid.service.ts]: Could change it to:
```
               (res) => {
                 this.log.d(`Bid placed for hash=${listing.hash} shipping to addressId=${addressIdOfProfile}`);
                 if (++nBidsPlaced === cart.listings.length) {
                   observer.next(true);
                   observer.complete();
                 }
               },
               observer.error);
```

File: src/app/core/market/api/cart/cart.service.ts [hunk 2]
```diff
@@ -56,12 +56,9 @@ export class CartService {
   clearCart(): Observable<any> {
     this.log.d(`Clearing cart with id=1`);
     return this.market.call('cart', ['clear', 1]).do(
-      data => {
-        this.snackbar.open('All Items Cleared From Cart')
-        this.updateCart();
-      },
+      data => this.updateCart(),
```
Comment (rynomster) [inline src/app/core/market/api/cart/cart.service.ts]: could just be
`return this.market.call('cart', ['clear', 1]).do(this.updateCart, this.snackbar.open);`

File: src/app/market/buy/checkout-process/checkout-process.component.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+  /* cart */
+
+  goToListings(): void {
+    this.router.navigate(['/market/overview']);
+  }
+
+  removeFromCart(shoppingCartId: number): void {
+    this.cartService.removeItem(shoppingCartId).take(1)
+      .subscribe(res => this.getCart());
```
Comment (rynomster) [inline src/app/market/buy/checkout-process/checkout-process.component.ts]: `.subscribe(this.getCart);`

File: src/app/market/buy/checkout-process/checkout-process.component.ts [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    this.cartService.clearCart().subscribe(res => {
+      if (isSnack) {
+        this.snackbarService.open('All Items Cleared From Cart');
+      }
+      this.getCart()
+    });
+  }
+
+  getCart(): void {
+    this.cartService.getCart().take(1).subscribe(cart => {
```
Comment (rynomster) [inline src/app/market/buy/checkout-process/checkout-process.component.ts]: remove `{` and `}`

File: src/app/market/buy/checkout-process/checkout-process.component.ts [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+        console.log(profile);
+        const addresses = profile.ShippingAddresses;
+        if (addresses.length > 0) {
+          this.shippingFormGroup.patchValue(addresses[0]);
+        }
+      });
+  }
+
+  valueOf(field: string) {
+    if (this.shippingFormGroup) {
```
Comment (rynomster) [inline src/app/market/buy/checkout-process/checkout-process.component.ts]: `return this.shippingFormGroup ? this.shippingFormGroup.get(field).value : ''`

**Comment by anandsinghparihar** [comment]
@rynomster complete all requested changes

**Pull request merged**